packages/workspace/src/utils/cli-config-utils.ts

```diff
@@ -39,7 +39,11 @@ export function replaceAppNameWithPath(
       `([^a-z0-9]*(${appName}))|((${appName})[^a-z0-9:]*)`,
       'gi'
     );
-    if (!!node.match(matchPattern)) {
+    if (
+      !!node.match(matchPattern) &&
+      node !== 'application' &&
+      node !== 'library'
```

Comment:
The problem with this is on line 47 where if the app name is included in the node, then we replace it with the root. For example, say the node was `application` and the appName was `app`; the `!!node.match(matchPattern)` would return true since `app` is included in `application`. Then on line 47 the `app` in `application` is replaced with the root (`apps/app`) and becomes `apps/application`.

Checking that the node is not equal to `application` (or `library`, since it has the same issue) gets around this issue. I think there might be an issue here still if someone names the app `application` but wasn't sure if I should take that rather edge case into consideration.